### ISSUE TITLE
Complain about redundant error checks before return

### ISSUE THREAD

**Issue by abbot** [golang/lint#312]
Golint should complain about this construct:

```
if err := foo(); err != nil {
  return err
}
return nil
```

and suggest to replace it with

```
return foo()
```

**Comment by abbot** [comment]
Fixed in #319.
